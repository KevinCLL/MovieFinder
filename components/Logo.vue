<template>
  <div class="m-5 mb-12">
    <h1 class="mb-3">
      <nuxt-link to="/">
        <img class="logo" src="~/assets/img/logo.png" alt="MovieFinder" />
      </nuxt-link>
    </h1>
    <h2 class="text-xl lg:text-3xl italic">
      Not <span class="text-red-700">the best movie finder</span>, but it's good
      enough <span class="text-xs">... I hope</span>
    </h2>
  </div>
</template>

<style>
.logo {
  animation: 1s appear;
  margin: auto;
}

@keyframes appear {
  0% {
    opacity: 0;
  }
  50% {
    transform: scale(1.5);
  }
  to {
    transform: scale(1);
  }
}
</style>
