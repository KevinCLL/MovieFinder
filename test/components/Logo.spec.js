import { shallowMount, createLocalVue, RouterLinkStub } from '@vue/test-utils'
import Logo from '@/components/Logo.vue'

const localVue = createLocalVue()

describe('Logo', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(Logo, {
      localVue,
      stubs: {
        NuxtLink: RouterLinkStub,
      },
    })
  })
  afterEach(() => {
    if (wrapper) {
      wrapper.destroy()
    }
  })

  test('is a Vue instance', () => {
    expect(wrapper.vm).toBeTruthy()
  })

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
