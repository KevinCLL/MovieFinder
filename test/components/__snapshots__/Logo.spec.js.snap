// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logo renders correctly 1`] = `
<div
  class="m-6 mb-12 text-center"
>
  <h1
    class="mb-3"
  >
    <a>
      <img
        alt="MovieFinder"
        class="logo"
        src="~/assets/img/logo.png"
      />
    </a>
  </h1>
   
  <h2
    class="text-xl lg:text-3xl italic"
  >
    
    Not 
    <span
      class="text-red-700"
    >
      the best movie finder
    </span>
    , but maybe
    it's good enough.
  
  </h2>
</div>
`;
